solar_system.rb

```diff
@@ -0,0 +1,63 @@
+class SolarSystem
+  attr_reader :star_name, :planets
+
+  def initialize(star_name)
+    @star_name = star_name
+    @planets = []
+  end
+
+  def add_planet(planet)
+    @planets << planet
+  end
+
+  def list_planets
+    output = "\nPlanets orbiting #{@star_name}\n"
+
+    @planets.each_with_index do |planet, index|
+      output = output + "#{index + 1}. #{planet.name}\n"
+    end
+
+    return output
+  end
+
+  def find_planet_by_name(string)
+    found_planet = @planets.find_all { |planet| planet.name.downcase == string.downcase }
+
+    if found_planet.length > 1
+      puts "Hm, we found multiple planets by that name. Maybe you can figure it out?"
+      puts found_planet
+      return found_planet
+    else
+      return found_planet[0]
+    end
+  end
+
+  def distance_between
+    puts "\nPls tell me the names of the two planets you would like to find the distance between"
+    puts "First planet:"
+    planet1 = gets.chomp
```

Comment:
I would break this into two methods. In `main.rb` I would have a method that does the work of interacting with the user, getting the two planet names. It then passes those names as arguments into an instance method of `SolarSystem` that finds the corresponding instances of planet and returns the distance between them, which the first method prints out.

The key observation is that "interact with the user" and "calculate the distance between two planets" are two separate tasks.